src/jit/objectalloc.h

```diff
@@ -52,32 +72,89 @@ inline ObjectAllocator::ObjectAllocator(Compiler* comp)
     : Phase(comp, "Allocate Objects", PHASE_ALLOCATE_OBJECTS)
     , m_IsObjectStackAllocationEnabled(false)
     , m_AnalysisDone(false)
+    , m_bitVecTraits(comp->lvaCount, comp)
+    , m_HeapLocalToStackLocalMap(comp->getAllocator())
 {
     // Disable checks since this phase runs before fgComputePreds phase.
     // Checks are not expected to pass before fgComputePreds.
-    doChecks = false;
+    doChecks                   = false;
+    m_EscapingPointers         = BitVecOps::UninitVal();
+    m_ConnGraphAdjacencyMatrix = nullptr;
 }
 
+//------------------------------------------------------------------------
+// IsObjectStackAllocationEnabled: Returns true iff object stack allocation is enabled
+//
+// Return Value:
+//    Returns true iff object stack allocation is enabled
+
 inline bool ObjectAllocator::IsObjectStackAllocationEnabled() const
 {
     return m_IsObjectStackAllocationEnabled;
 }
 
+//------------------------------------------------------------------------
+// EnableObjectStackAllocation:       Enable object stack allocation.
+
 inline void ObjectAllocator::EnableObjectStackAllocation()
 {
     m_IsObjectStackAllocationEnabled = true;
 }
 
 //------------------------------------------------------------------------
-// CanAllocateLclVarOnStack: Returns true iff local variable can not
-//                           potentially escape from the method and
-//                           can be allocated on the stack.
-inline bool ObjectAllocator::CanAllocateLclVarOnStack(unsigned int lclNum) const
+// CanAllocateLclVarOnStack: Returns true iff local variable can be
+//                           allocated on the stack.
+//
+// Arguments:
+//    lclNum   - Local variable number
+//    clsHnd   - Class handle of the variable class
+//
+// Return Value:
+//    Returns true iff local variable can be allocated on the stack.
+//
+// Notes:
+//    Stack allocation of objects with gc fields and boxed objects is currently disabled.
+
+inline bool ObjectAllocator::CanAllocateLclVarOnStack(unsigned int lclNum, CORINFO_CLASS_HANDLE clsHnd)
+{
+    assert(m_AnalysisDone);
+
+    if (comp->info.compCompHnd->getClassAttribs(clsHnd) & CORINFO_FLG_CONTAINS_GC_PTR)
+    {
+        // TODO-ObjectStackAllocation: enable stack allocation of objects with gc fields
+        return false;
+    }
+
+    if (comp->info.compCompHnd->isValueClass(clsHnd))
```

Comment:
Since you've already paid the cost to fetch the class attributes above you can just check for `CORINFO_FLG_VALUECLASS` here.

Reply:
Done.